lib/instances/BindableFunction.lua

```diff
@@ -0,0 +1,52 @@
+local BaseInstance = import("./BaseInstance")
+local InstanceProperty = import("../InstanceProperty")
+
+local BindableFunction = BaseInstance:extend("BindableFunction", {
+	creatable = true,
+})
+
+-- This isn't type checked because Roblox doesn't.
+-- Instead, you just get a "attempt to call a [whatever] value" error when you :Invoke()
+BindableFunction.properties.OnInvoke = InstanceProperty.writeOnly({
+	getDefault = function()
+		return nil
+	end,
+})
+
+function BindableFunction.prototype:Invoke(...)
+    local callback = getmetatable(self).instance.properties.OnInvoke
+
+    -- BindableFunctions DON'T error if OnInvoke is nil, unlike every other type.
+    if callback == nil then
+        callback = function() end -- noop
+    end
+
+    -- Replicate Roblox behavior in that mixed tables will only send the numerical part
+    local arguments = {...}
+
+    for index,value in ipairs(arguments) do
+        if type(value) == "table" then
+            -- Check if a mixed table
+            local numIndexes, numNamedKeys = #value, 0
+
+            for _ in pairs(value) do
+                numNamedKeys = numNamedKeys + 1
+            end
+
+            if numIndexes > 0 and numNamedKeys ~= numIndexes then
+                -- Mixed table, only send numerical portion
+                local newValue = {}
+
+                for _,value2 in ipairs(value) do
+                    table.insert(newValue, value2)
+                end
+
+                arguments[index] = newValue
+            end
+        end
+    end

```

Comment:
There are two issues with this function right now. Nested tables won't have the correct treatment applied to them, and we need to be careful with object identity.

For example:

```lua
local x = {
    value = 5
}
bindable:Invoke(x)
x.value = 6
-- Code running inside `OnInvoke` can observe that `x.value` has changed
```

Mutating `x` should not affect the code running on the other side of the bindable, but unless we explicitly deep-copy all tables, that's what will happen.

Reply:
This is a really tricky function to write. Have any ideas *without* deep copying all the tables?